src/main/java/com/google/api/codegen/transformer/ruby/RubyGapicSurfaceDocTransformer.java

```diff
@@ -0,0 +1,78 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer.ruby;
+
+import com.google.api.codegen.ProtoFileView;
+import com.google.api.codegen.config.ApiConfig;
+import com.google.api.codegen.gapic.GapicCodePathMapper;
+import com.google.api.codegen.transformer.FileHeaderTransformer;
+import com.google.api.codegen.transformer.GrpcElementDocTransformer;
+import com.google.api.codegen.transformer.ModelToViewTransformer;
+import com.google.api.codegen.transformer.ModelTypeTable;
+import com.google.api.codegen.transformer.SurfaceNamer;
+import com.google.api.codegen.util.ruby.RubyTypeTable;
+import com.google.api.codegen.viewmodel.GrpcDocView;
+import com.google.api.codegen.viewmodel.ImportSectionView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.ProtoFile;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+
+public class RubyGapicSurfaceDocTransformer implements ModelToViewTransformer {
+  private static final String DOC_TEMPLATE_FILENAME = "ruby/message.snip";
+
+  private final GapicCodePathMapper pathMapper;
+  private final FileHeaderTransformer fileHeaderTransformer = new FileHeaderTransformer(null);
+  private final GrpcElementDocTransformer elementDocTransformer = new GrpcElementDocTransformer();
+
+  public RubyGapicSurfaceDocTransformer(GapicCodePathMapper pathMapper) {
+    this.pathMapper = pathMapper;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return ImmutableList.of(DOC_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    ImmutableList.Builder<ViewModel> surfaceDocs = ImmutableList.builder();
+    for (ProtoFile file : new ProtoFileView().getElementIterable(model)) {
+      surfaceDocs.add(generateDoc(file, apiConfig));
+    }
+    return surfaceDocs.build();
+  }
+
+  private ViewModel generateDoc(ProtoFile file, ApiConfig apiConfig) {
+    String subPath = pathMapper.getOutputPath(file, apiConfig);
+    String protoFilename = file.getSimpleName();
+    String baseFilename =
+        protoFilename.substring(0, protoFilename.length() - "proto".length()) + "rb";
+    ModelTypeTable typeTable =
+        new ModelTypeTable(
+            new RubyTypeTable(apiConfig.getPackageName()),
+            new RubyModelTypeNameConverter(apiConfig.getPackageName()));
+    SurfaceNamer namer = new RubySurfaceNamer(typeTable.getFullNameFor(file));
```

Comment:
The docs for the files are not in the same package as the main API. You end up with docs in the tests for `Library::V1::Timestamp` instead of `Google::Protobuf::Timestamp`.